### PR TITLE
fix(@desktop/mentions): editing mentions

### LINKED ISSUE #3543: When going back and editing an @mention, the @mention autocomplete should open
Currently it doesn't open, this is a bug

### PR DESCRIPTION
**Pull request by pinebit**
Fixes #3543

### CODE REVIEW

File: ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml [hunk 1]
```diff
@@ -12,26 +12,33 @@ Item {
...
         if (!isFilteringPropertyOk())
             return
 
-        var filter = getFilter()
-        if (!isFilterReady(filter))
+        let filter = getFilter()
+        if (filter === undefined)
+            return
+
+        this.lastAtPosition = -1
```
Comment (alaibe) [inline ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml:34]: in this context, wouldn't be more clear to not use `this` and pass lastAtPosition as a normal parameter?

File: ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml [hunk 2]
```diff
@@ -69,41 +76,19 @@ Item {
...
-        // Prevents suggestions to show up at all
-        if (filter.indexOf("@") === -1)  {
-          return false
-        }
-
-        let hasWhiteSpaceBeforeCursor = filter.charAt(this.cursorPosition - 1) === " "
-
-        if (filter.charAt(this.cursorPosition - 2) === "@" && hasWhiteSpaceBeforeCursor) {
-            return false
-        }
```
Comment (alaibe) [inline ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml:93]: I don't think I can see this condition in the new change but I am also not sure if this is a valid one!

File: ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml (hunk 1 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml:34]: This is used as a property from outside of this component.
Overall design of this thing is awkward, I would wish to rebuild it, but I wanted to fix the bug in the first place.

File: ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml (hunk 2 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml:93]: I found this condition very suspicious and error-prone, so I removed it. Then I checked if the feature is not broken and it looked good. So I decided to keep this removed.

**Review by alaibe — APPROVED** [review]

File: ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml (hunk 2 above)
Comment (alaibe) [reply ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml:93]: Yeah I had similar feeling when reading it but as you tested and looks good, it seems legit!

File: ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml (hunk 1 above)
Comment (alaibe) [reply ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml:34]: Agree, I cross the path of this SuggestionBox too and I was quite confuse by it.

**Pull request merged**